src/jloptions.c

```diff
@@ -505,6 +507,20 @@ JL_DLLEXPORT void jl_parse_opts(int *argcp, char ***argvp)
             else
                 jl_errorf("julia: invalid argument to --handle-signals (%s)", optarg);
             break;
+        case 'r':
```

Comment:
I don't know what's exactly the use case so I'm fine with an option. Just note that this will not propagate automatically to workers or other processes you starts.